pkg/operator/config.go

```diff
@@ -0,0 +1,163 @@
+	Controller      string `json:"images"`
+}
+
+// Images allows build systems to inject images for MAO components
+type Images struct {
+	ClusterAPIControllerAWS       string `json:"clusterAPIControllerAWS"`
+	ClusterAPIControllerOpenStack string `json:"clusterAPIControllerOpenStack"`
+	ClusterAPIControllerLibvirt   string `json:"clusterAPIControllerLibvirt"`
+}
+
+// InstallConfig contains the mao relevant config coming from the install config, i.e provider
+type InstallConfig struct {
+	IntallPlatform `json:"platform"`
+}
+
+// Platform is the configuration for the specific platform upon which to perform
+// the installation. Only one of the platform configuration should be set
+type IntallPlatform struct {
+	// AWS is the configuration used when running on AWS
+	AWS interface{} `json:"aws,omitempty"`
+
+	// Libvirt is the configuration used when running on libvirt
+	Libvirt interface{} `json:"libvirt,omitempty"`
+
+	// OpenStack is the configuration used when running on OpenStack
+	OpenStack interface{} `json:"openstack,omitempty"`
+}
+
+func getInstallConfig(client kubernetes.Interface) (*InstallConfig, error) {
+	cm, err := client.CoreV1().ConfigMaps(ClusterConfigNamespace).Get(ClusterConfigName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed getting clusterconfig %s/%s: %v", ClusterConfigNamespace, ClusterConfigName, err)
+	}
+
+	return getInstallConfigFromClusterConfig(cm)
+}
+
+// UnmarshalInstallConfig builds an install config from the cluster config.
+func getInstallConfigFromClusterConfig(clusterConfig *corev1.ConfigMap) (*InstallConfig, error) {
+	icYaml, ok := clusterConfig.Data[InstallConfigKey]
+	if !ok {
+		return nil, fmt.Errorf("missing %q in configmap", InstallConfigKey)
+	}
+	var ic InstallConfig
+	if err := yaml.Unmarshal([]byte(icYaml), &ic); err != nil {
+		return nil, fmt.Errorf("invalid InstallConfig: %v\nyaml:\n%s", err, icYaml)
+	}
+	return &ic, nil
+}
+
+func getProviderFromInstallConfig(installConfig *InstallConfig) (Provider, error) {
+	v := reflect.ValueOf(installConfig.IntallPlatform)
+	var nonNilFields int
+
+	for i := 0; i < v.NumField(); i++ {
+		if v.Field(i).Interface() != nil {
+			nonNilFields = nonNilFields+1
+		}
+		if nonNilFields > 1 {
+			return "", fmt.Errorf("more than one platform provider given")
+		}
+	}
+
+	if installConfig.AWS != nil {
+		return AWSProvider, nil
+	}
+	if installConfig.Libvirt != nil {
+		return LibvirtProvider, nil
+	}
+	if installConfig.OpenStack != nil {
+		return OpenStackProvider, nil
+	}
+	return "", fmt.Errorf("no platform provider found on install config")
+}
+
+func getImagesFromJSONFile(filePath string) (*Images, error) {
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var i Images
+	if err := json.Unmarshal(data, &i); err != nil {
+		return nil, err
+	}
+	return &i, nil
+}
+
+func getProviderControllerFromImages(provider Provider, images Images) (string, error) {
+	switch provider{
```

Comment:
This looks badly formatted. Would expect `gofmt` to leave a space before `{`.

Reply:
Why doesn't CI pick this up? @paulfantom

Reply:
indeed, is gofmt broken on CI? cc @paulfantom